utility/sosreport.py

```diff
@@ -0,0 +1,127 @@
+"""Standard script to collect all the logs from ceph cluster through and place it in directory provided
+
+Through installer node get all other nodes in the cluster, generate sosreport for all the nodes obtained.
+then upload all the collected logs to magna/directory(run_dir) provided
+
+  Typical usage example:
+
+  python sosreport.py --ip x.x.x.x --username abc --password abcd --directory /tmp/cephci-run-ysfyu
+  python sosreport.py -h
+"""
+import os
+import re
+import sys
+
+from docopt import docopt
+
+from ceph.ceph import SSHConnectionManager
+from ceph.parallel import parallel
+
+doc = """
+Standard script to collect all the logs from ceph cluster
+
+    Usage:
+        sosreport.py --ip <str> --username <str> --password <str> --directory <str>
+        sosreport.py (-h | --help)
+
+    Options:
+        -h --help          Shows the command usage
+        --ip <str>         IP address of Installer node.
+        --username <str>   Username to be used to access the system other than root
+        --password <str>   password of given username
+        --directory <str>  directory/folder
+"""
+
+
+def generate_sosreport_in_node(
+    nodeip: str, uname: str, pword: str, directory: str, results: list
+) -> None:
+    """Generate sosreport in the given node and copy report to directory provided
+
+    Args:
+       nodeip              host Ip address
+       uname               Username for accessing host
+       pword               password for accessing host through given user
+       directory           directory to store all the logs
+       results             host IP address for which this operation are failed
+    """
+    print(f"Connecting {nodeip} to generate sosreport")
+    try:
+        ssh_d = SSHConnectionManager(nodeip, uname, pword).get_client()
+        ssh_d.exec_command("sudo yum -y install sos")
+        stdin, stdout, stderr = ssh_d.exec_command(
+            "sudo sosreport -a --all-logs -e ceph --batch"
+        )
+        sosreport = re.search(r"sosreport-.*.tar.xz", stdout.read().decode())
+        ssh_d.exec_command(f"sudo chmod 755 /var/tmp/{sosreport.group()}")
```

Comment:
Let's check if we have results in `sosreport` by using `if not sosreport: return`

Instead of `chmod` can we use `chown cephuser:cephuser /var/tmp/{sosreport.group()}`

Reply:
Made changes